src/test/java/spoon/test/ctStatementList/CtStatementListTest.java

```diff
@@ -1,35 +1,52 @@
+/**
+ * Copyright (C) 2006-2021 INRIA and contributors
+ * Spoon - http://spoon.gforge.inria.fr/
+ *
+ * This software is governed by the CeCILL-C License under French law and
+ * abiding by the rules of distribution of free software. You can use, modify
+ * and/or redistribute the software under the terms of the CeCILL-C license as
+ * circulated by CEA, CNRS and INRIA at http://www.cecill.info.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the CeCILL-C License for more details.
+ *
+ * The fact that you are presently reading this means that you have had
+ * knowledge of the CeCILL-C license and that you accept its terms.
+ */
 package spoon.test.ctStatementList;
 
 import org.junit.jupiter.api.Test;
 import spoon.Launcher;
 import spoon.reflect.code.CtStatement;
 import spoon.reflect.code.CtStatementList;
 import spoon.reflect.factory.Factory;
-import spoon.support.reflect.code.CtStatementListImpl;
 
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.CoreMatchers.is;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
 public class CtStatementListTest {
+    private static Factory factory;
+
+    private static CtStatementList getStatementListInitializedWithOneStatement() {
+        factory = new Launcher().getFactory();
```

Comment:
Setting the `factory` field is a very unexpected side-effect of this method. I would in general avoid mutable fields in test classes (or any classes, for that matter) when possible. If a new test is added that uses this field without first calling the associated method, there could be an NPE if that test is executed first. Tests that depend on the order of test execution is a big nono.

If you really want this as a field, then set it up in an `@BeforeEach` method, but make it an instance field and not static. I'd personally avoid even that, though, and just create the factory in each test that needs it.

Just create the factory in each test instead.

```suggestion
    private static CtStatementList getStatementListInitializedWithOneStatement() {
        Factory factory = new Launcher().getFactory();
```

Reply:
> Just create the factory in each test instead.

Thanks, I will do the same